xarray/plot/dataset_plot.py

```diff
@@ -344,13 +367,34 @@ def newplotfunc(
                 cbar_kwargs["label"] = meta_data.get("hue_label", None)
             _add_colorbar(primitive, ax, cbar_ax, cbar_kwargs, cmap_params)
 
+        if meta_data["add_quiverkey"]:
+            magnitude = _get_nice_quiver_magnitude(ds[u], ds[v])
+            units = ds[u].attrs.get("units", "")
```

Comment:
could you test the unit string of u is used?

Reply:
done.